user-service/src/main/java/rs/edu/raf/banka1/services/implementations/EmployeeServiceImpl.java

```diff
@@ -226,6 +236,42 @@ public NewPasswordResponse setNewPassword(String token, String password) {
         return new NewPasswordResponse(employee.getUserId());
     }
 
+    @Override
+    public void resetLimitForEmployee(Long employeeId) {
+        Employee employee = employeeRepository.findById(employeeId).orElseThrow(() -> new EmployeeNotFoundException(employeeId));
+        employee.setLimitNow(0.0);
+        employeeRepository.save(employee);
+    }
+
+    @Override
+    public void resetEmployeeLimits() {
+        if(LocalDate.now().getDayOfWeek().equals(DayOfWeek.SATURDAY) ||
+                LocalDate.now().getDayOfWeek().equals(DayOfWeek.SUNDAY)) {
+            return;
+        }
+        List<Employee> users = employeeRepository.findAll();
+        users.forEach(user->user.setLimitNow(0.0));
+        employeeRepository.saveAll(users);
+    }
+
+    @Override
+    public LimitDto setLimitForEmployee(NewLimitDto newLimitDto) {
+        Long employeeId = newLimitDto.getUserId();
+        Employee employee = employeeRepository.findById(employeeId).orElseThrow(()-> new EmployeeNotFoundException(employeeId));
+        if(!employee.getPosition().equalsIgnoreCase(Constants.AGENT)) {
+            throw new ForbiddenException("Employee with id: " + employeeId + " is not in agent position. Changing the limit is prohibited.");
+        }
+        employee.setOrderlimit(newLimitDto.getLimit());
+        employee.setRequireApproval(newLimitDto.getApprovalRequired());
+        Employee saved = employeeRepository.save(employee);
+        return limitMapper.toLimitDto(employee);
+    }
+
+    @Override
+    public List<LimitDto> getAllLimits(Employee currentAuth) {
+        return this.employeeRepository.findAll().stream().map(limitMapper::toLimitDto).collect(Collectors.toList());
+    }
+
     @Override
     public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
         Optional<Employee> myEmployee = this.employeeRepository.findByEmail(username);
```

Comment:
zašto nam je findByEmail za username?

Reply:
Mislim da Spring Security ne razlikuje email i username (vise je do implementacije), ali to predstavlja unique field koji ce pored userId-a identifikovati korisnika (na primer pri loginu). Odnosno mora da se zove username po library-u, ali predstavlja email